python/cudf/cudf/_lib/scalar.pyx

```diff
@@ -178,7 +178,7 @@ cdef class DeviceScalar:
         return self.get_raw_ptr()[0].is_valid()
 
     def __repr__(self):
-        if self.value is NA:
+        if self.value is NA or self.value is NaT:
```

Comment:
Do we want to have a helper function like `is_na_like` (to avoid confusion with the pandas name `isna`)? That would help reduce repeated implementation of the same logic. We also have some places where we write `x in {None, NA, NaT}` and other places where we write `x is None or x is NA or x is NaT`. Those seem like they should be cleaned up and use standard/common logic in this PR -- maybe a couple flavors of that logic, depending on `None` treatment.

Reply:
Yup, added `is_na_like` utility to remove this common pattern repetition.